src/main/java/com/greenfox/poker/service/UserService.java

```diff
@@ -2,77 +2,114 @@
+  @Autowired
+  PokerUserRepo pokerUserRepo;
+
+  @Autowired
+  PokerUserDTO pokerUserDTO;
+
+  @Autowired
+  TokenService tokenService;
+
+  public ResponseType responseToSuccessfulRegister(PokerUser pokerUser) {
+    pokerUserRepo.save(pokerUser);
+    String token = tokenService.generateToken(pokerUser);
+    return new UserTokenResponse("success", token, pokerUser.getId());
   }
 
-  public ResponseEntity<?> respondToMissingParameters(BindingResult bindingResult) {
-    List<String> listOfMissingFields = new ArrayList<>();
+  public ResponseType responseToSuccessfulLogin(LoginRequest loginRequest) {
+    PokerUser pokerUserFromDatabase = pokerUserRepo.findByUsername(loginRequest.getUsername()).get(0);
+    String token = tokenService.generateToken(pokerUserFromDatabase);
+    return new UserTokenResponse("success", token, pokerUserFromDatabase.getId());
+  }
+
+  public ResponseType respondToMissingParameters(BindingResult bindingResult) {
+    List<String> missing = new ArrayList<>();
     String missingFields = new String();
     for (FieldError fielderror : bindingResult.getFieldErrors()) {
-      listOfMissingFields.add(fielderror.getField());
+      missing.add(fielderror.getField());
     }
-    missingFields = "Missing parameter(s): " + listOfMissingFields.toString();
-    return new ResponseEntity(new StatusError("fail", missingFields), HttpStatus.BAD_REQUEST);
+    System.out.println(missingFields);
+    missingFields = "Missing parameter(s): " + missing.stream().collect(Collectors.joining(", ")) + "!";
+    return new StatusError("fail", missingFields);
   }
 
-  public ResponseEntity<?> respondToOccupiedConflict(String errorMessage) {
-    StatusError statusError = new StatusError("fail", errorMessage);
-    return new ResponseEntity(statusError, HttpStatus.CONFLICT);
+  public ResponseType registerWithOccupiedEmail() {
+
+    return new StatusError("fail", "email address already exists");
   }
 
-  public ResponseEntity<?> respondToInvalidUsernameOrPassword(String errorMessage) {
-    StatusError statusError = new StatusError("fail", errorMessage);
-    return new ResponseEntity(statusError, HttpStatus.UNAUTHORIZED);
+  public ResponseType registerWithOccupiedUsername() {
+    return new StatusError("fail", "username already exists");
   }
 
-  public ResponseEntity<?> register(PokerUser pokerUser,
-          BindingResult bindingResult) {
-    if (bindingResult.hasErrors()) {
-      return respondToMissingParameters(bindingResult);
-    } else if (isEmailOccupied(pokerUser)) {
-      return respondToOccupiedConflict("email address already exists");
-    } else if (isUsernameOccupied(pokerUser)) {
-      return respondToOccupiedConflict("username already exists");
+  public boolean isLoginValid(LoginRequest loginRequest){
+    List<PokerUser> users = pokerUserRepo.findByUsername(loginRequest.getUsername());
+    if (users.size() > 0 && users.get(0).getPassword().equals(loginRequest.getPassword())){
```

Comment:
For sake of easier debugging, please separate this method to multiple lines.
Get the size of your users list,
get the password,
and then compare these in this condition.